RxSwiftWorkshop/Maps/ViewModel/FlightVisualizationViewModel.swift

```diff
@@ -0,0 +1,60 @@
+//
+// Created by Krzysztof Siejkowski on 03/05/2017.
+// Copyright (c) 2017 Mobile Academy. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+import CoreLocation
+
+protocol FlightVisualisation {
+    var route: Observable<Route> { get }
+}
+
+final class FlightVisualisationViewModel: FlightVisualisation {
+
+    enum Constants {
+        static let amsterdamLocation = CLLocation(latitude: 52.310475, longitude: 4.768158)
+    }
+
+    enum Error: Swift.Error {
+        case notEnoughInformationToVisualiseFlight
+    }
+
+    private let geolocator: Geolocator
+    private let geocodingCache: Observable<CLLocation>
+    private let flight: Flight
+    private let formatter = AirportFormatter()
+
+    private let internalRoute = PublishSubject<Route>()
+    var route: Observable<Route> {
+        return geocodingCache.flatMap { [weak self] location -> Observable<Route> in
```

Comment:
Och, I see that's more difficult. `Variable` wouldn't work here.
Maybe, just maybe, that error logic could be extracted elsewhere.